vyper/semantics/analysis/local.py

```diff
@@ -242,7 +233,19 @@ def visit_AnnAssign(self, node):
             self.namespace[name] = VarInfo(type_, location=DataLocation.MEMORY)
         except VyperException as exc:
             raise exc.with_annotation(node) from None
-        self.expr_visitor.visit(node.value)
+
+        self.expr_visitor.visit(node.target, type_)
+        self.expr_visitor.visit(node.value, type_)
+
+    def visit_Assert(self, node):
+        if node.msg:
+            _validate_revert_reason(node.msg)
+
+        try:
+            validate_expected_type(node.test, BoolT())
+        except InvalidType:
+            raise InvalidType("Assertion test value must be a boolean", node.test)
```

Comment:
i don't think we need to change the exception message here - without this catch clause, we get something like the following (which i think is reasonable):
```vyper
Error compiling: tmp/assert.vy
vyper.exceptions.InvalidType: Expected bool but literal can only be cast as int104 or uint96.
  contract "tmp/assert.vy:3", function "foo", line 3:11 
       2 def foo():
  ---> 3     assert 1, "hello"
  ------------------^
       4
```

Reply:
(looks like this was already present in the old version of `local.py`)